servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DynamicCompositeCancellable.java

```diff
@@ -34,8 +34,9 @@
      * Add a {@link Cancellable} that will be cancelled when this object's {@link #cancel()} method is called,
      * or be cancelled immediately if this object's {@link #cancel()} method has already been called.
      * @param toAdd The {@link Cancellable} to add.
-     * @return {@code true} if the {@code toAdd} was added, and {@code false} if {@code toAdd} was not added and
-     * {@link Cancellable#cancel()} was called on {@code toAdd}.
+     * @return {@code true} if the {@code toAdd} was added, and {@code false} if {@code toAdd} was not added because
+     * it already exists. If {@code false} then {@link Cancellable#cancel()} will be called unless the reason is this
```

Comment:
note this interface is internal, but the semantics around it didn't really make sense for `set` so I clarified here and added a unit test to verify.